ydb_sqlalchemy/sqlalchemy/__init__.py

```diff
@@ -326,6 +343,24 @@ def visit_function(self, func, add_to_result_map=None, **kwargs):
             + [name]
         ) % {"expr": self.function_argspec(func, **kwargs)}
 
+    def _yson_convert_to(self, statement: str, target_type: sa.types.TypeEngine) -> str:
+        type_name = target_type.compile(self.dialect)
+        if isinstance(target_type, sa.Numeric) and not isinstance(target_type, (sa.Float, sa.Double)):
```

Comment:
It looks strage:
Double, Float, String, etc - map to concrete YDB type, but we have special type set (for example Decimal) which will try to convert to any column type.

I think it cause errors, for example type to sum decimal values, presented as String.

What about direct support YDB Decimals for decimal.

If decimal stored as strings - from point of database it is string and decimal -> string conversion will be on client side (not in alchemy).

Reply:
In this particular place we handle the following situation:
```
table.c.json_data['my_field'].as_numeric(22, 9)
```
So the user asks database to extract that field from Json and convert it to Decimal(22, 9).

In other dialects there is a CAST(... AS Decimal(22, 9), because Json can not contain Decimal value. This CAST works fine with strings, integers and floats.

But in case of YDB the only type, which could be casted to Decimal, is String, so I extract a value (string, int or float) as string and do CAST.